examples/tgn.py

```diff
@@ -26,14 +27,17 @@
 
 path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', 'JODIE')
 dataset = JODIEDataset(path, name='wikipedia')
-data = dataset[0].to(device)
+data = dataset.data
 
 # Ensure to only sample actual destination nodes as negatives.
 min_dst_idx, max_dst_idx = int(data.dst.min()), int(data.dst.max())
-
-train_data, val_data, test_data = data.train_val_test_split(
+train_dataset, val_dataset, test_dataset = dataset.train_val_test_split(
     val_ratio=0.15, test_ratio=0.15)
 
+train_loader = TemporalDataLoader(train_dataset, batch_size=200)
```

Comment:
I liked it a bit more before. That is, `train_val_test_split` operates on single `data` objects. With this, `TemporalDataLoader` expects `train_data`, ...  rather than `train_dataset`, ...

Reply:
Yep, the previous implementation seems simpler. But, I wasn't able to get it working with a DataLoader ... (actually I've spent some hours trying to get it working). This implementation is also working with DataLoaders with multiple workers  (which wasn't possible with the previous one since it wasn't using DataLoader).

Reply:
Please see my comments above.